src/renderer/features/network/ManageCustomRpcNode/model/edit-custom-rpc-model.ts

```diff
@@ -0,0 +1,175 @@
+import { createEffect, createEvent, createStore, sample } from 'effector';
+import { createForm } from 'effector-forms';
+
+import { networkService } from '@shared/api/network';
+import { ExtendedChain } from '@entities/network';
+import { VerifyRpcConnectivityFxParams, EditRpcNodeFxParams, RpcConnectivityResult } from '../lib/custom-rpc-types';
+import { manageNetworkModel } from '@pages/Settings/Networks/model/manage-network-model';
+import { RpcNode } from '@shared/core';
+import { customRpcConstants } from '../lib/custom-rpc-constants';
+import { customRpcUtils } from '../lib/custom-rpc-utils';
+
+const formInitiated = createEvent();
+const flowStarted = createEvent();
+const flowFinished = createEvent();
+
+const networkChanged = createEvent<ExtendedChain>();
+const nodeSelected = createEvent<RpcNode>();
+const rpcConnectivityVerified = createEvent<RpcConnectivityResult>();
+
+const $editCustomRpcForm = createForm({
+  fields: {
+    name: {
+      init: '',
+      rules: customRpcConstants.FieldRules.name,
+    },
+    url: {
+      init: '',
+      rules: customRpcConstants.FieldRules.url,
+    },
+  },
+  validateOn: ['submit'],
+});
+
+const $selectedNode = createStore<RpcNode | null>(null);
+const $selectedNetwork = createStore<ExtendedChain | null>(null);
+const $rpcConnectivityResult = createStore<RpcConnectivityResult>(RpcConnectivityResult.INIT);
+
+const $isFlowStarted = createStore<boolean>(false);
+const $isLoading = createStore<boolean>(false);
+
+const verifyRpcConnectivityFx = createEffect(
+  async ({ chainId, url }: VerifyRpcConnectivityFxParams): Promise<RpcConnectivityResult> => {
+    const validationResult = await networkService.validateRpcNode(chainId, url);
+
+    return customRpcConstants.RpcValidationMapping[validationResult];
+  },
+);
+
+const editRpcNodeFx = createEffect(({ network, form, nodeToEdit }: EditRpcNodeFxParams) => {
+  manageNetworkModel.events.rpcNodeUpdated({
+    chainId: network.chainId,
+    oldNode: nodeToEdit,
+    rpcNode: { url: form.url, name: form.name },
+  });
+});
+
+const updateInitialValuesFx = createEffect(({ url, name }: RpcNode) => {
+  $editCustomRpcForm.fields.url.onChange(url);
+  $editCustomRpcForm.fields.name.onChange(name);
```

Comment:
I am not sure that use store directly in effect is good idea, better to provide it through params. 

@tuul-wq correct me if I'm wrong